src/main/java/repositories/BookHandlerImpl.java

```diff
@@ -1,22 +1,23 @@
-package Data;
-
-import Library.Book;
-
+package repositories;
+import model.Book;
+import services.Statistics;
 import java.io.*;
 import java.util.*;
 
-public class BookHandler {
-    private String filePath = "C:\\Users\\david\\IdeaProjects\\LibraryProject\\Online-Library\\src\\main\\resources\\Data Base - List of Books.txt";
+public class BookHandlerImpl implements BookHandler {
+    private final String filePath = "C:\\Users\\david\\IdeaProjects\\LibraryProject\\Online-Library\\src\\main\\resources\\Data Base - List of Books.txt";
 
+    @Override
     public void saveBookInfo(Book book) {
         try (PrintWriter writer = new PrintWriter(new FileWriter(filePath, true))) {
             String formattedEntry = String.format("\"%s\" %s - %s", book.getTitle(), book.getAuthor(), book.getBookStatus());
             writer.println(formattedEntry);
         } catch (IOException exception) {
-            System.out.println("File not found");
+            throw new FilePathNotFoundException("An error occurred while saving the book information. File path not found.", exception);
         }
     }
 
+    @Override
     public List<Book> readAllBooks() { // this method reads the book data from a file
```

Comment:
What do you think about reading Books only once? Reading from file is pretty expensive operation. You can read it once when program is starting(or when user is logged in) than you can keep it in List of Books :)

Reply:
Pending... Need to discuss at the lesson